fedimint-server/src/config.rs

```diff
@@ -67,6 +70,8 @@ pub struct ServerConfigPrivate {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ServerConfigConsensus {
+    /// SHA256 hash of the consensus config values, for validation with other peers
+    pub hash: sha256::Hash,
```

Comment:
I feel like it should be cached outside that struct so that the struct itself can be hashed in full?